### PR TITLE
[SPARK-14279][BUILD] Pick the spark version from pom

### PR DESCRIPTION
**Pull request by dhruve**
Changes are already merged in master (https://github.com/apache/spark/pull/13061) . This one's for branch-2.0

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/package.scala [hunk 1]
```diff
@@ -41,7 +41,58 @@ package org.apache
  * level interfaces. These are subject to changes or removal in minor releases.
  */
 
+import java.util.Properties
+
 package object spark {
-  // For package docs only
-  val SPARK_VERSION = "2.0.0-preview"
+
+  private object SparkBuildInfo {
```
Comment (dhruve) [inline core/src/main/scala/org/apache/spark/package.scala]: @vanzin 

With the current state being in the preview mode, users testing the framework would see the version as 2.0.0-SNAPSHOT as being mentioned in the pom file.

If this is acceptable, rest of the changes are as is.

**Comment by rxin** [comment]
hm maybe no need to do this for branch-2.0, since it's pretty easy to do this manually once or twice for 2.0.x.

**Comment by vanzin** [comment]
Dhruve points out above an instance where a previous release wasn't properly changed and now Spark is printing the wrong version in branch-2.0; so I think this change does help.

**Comment by vanzin** [comment]
ok to test

**Comment by tgravescs** [comment]
Personally I would like to see this go into branch-2.0 also. it doesn't seem very risky but has benefits to both dev and end users.  @rxin do you have strong objections?

**Comment by rxin** [comment]
sounds good go ahead

**Comment by vanzin** [comment]
Merging to 2.0.

**Comment by vanzin** [comment]
@dhruve could you close the PR? The bot doesn't do it automatically for backports. thx

**Comment by dhruve** [comment]
@vanzin. Thanks for reviewing & merging the PR. Have closed it.

**Comment by chenghao-intel** [comment]
`spark-version-info.properties` cannot be found in my develop machine, and will cause NPE while debugging with IDE, should we add the default version info back for developers like me?

**Comment by tgravescs** [comment]
@chenghao-intel  can you file a jira with the details on what you are doing and link to the original jira?

**Pull request closed without merging**